src/romanToInteger/Solution.java

```diff
@@ -18,100 +20,59 @@ public int romanToInt(String s) {
-                    }
+                    nrobase = 50;
+                    descuento = 20;
+                    anteriorRoman = 'X';
+                    Converts valueL = new Converts();
+                    int digitL = valueL.evaluation(romans, anteriorRoman, i, nrobase, descuento);
+                    number = number + digitL;
+                    break;
 
                 case "C":
-                    if (i == 0) {
-                        number += 100;
-                        break;
-                    } else {
-                        j = i - 1;
-                        anteriorRoman = romans[j];
-                        if (anteriorRoman == 'X') {
-                            number += 100 - 20;
-                            break;
-                        } else {
-                            number += 100;
-                            break;
-                        }
-                    }
+                    nrobase = 100;
+                    descuento = 20;
+                    anteriorRoman = 'X';
+                    Converts valueC = new Converts();
+                    int digitC = valueC.evaluation(romans, anteriorRoman, i, nrobase, descuento);
+                    number = number + digitC;
+                    break;
 
                 case "D":
-                    if (i == 0) {
-                        number += 500;
-                        break;
-                    } else {
-                        j = i - 1;
-                        anteriorRoman = romans[j];
-                        if (anteriorRoman == 'C') {
-                            number += 500 - 200;
-                            break;
-                        } else {
-                            number += 500;
-                            break;
-                        }
-                    }
+                    nrobase = 500;
+                    descuento = 200;
+                    anteriorRoman = 'C';
+                    Converts valueD = new Converts();
+                    int digitD = valueD.evaluation(romans, anteriorRoman, i, nrobase, descuento);
+                    number = number + digitD;
+                    break;
 
                 case "M":
-                    if (i == 0) {
-                        number += 1000;
-                        break;
-                    } else {
-                        j = i - 1;
-                        anteriorRoman = romans[j];
-                        if (anteriorRoman == 'C') {
-                            number += 1000 - 200;
-                            break;
-                        } else {
-                            number += 1000;
-                            break;
-                        }
-                    }
+                    nrobase = 1000;
+                    descuento = 200;
+                    anteriorRoman = 'C';
+                    Converts valueM = new Converts();
+                    int digitM = valueM.evaluation(romans, anteriorRoman, i, nrobase, descuento);
```

Comment:
I suggest to extract (move) these lines to it own method.

Ideally, the whole switch statement should look like:

```java
switch (character) {
                case <case1>:
                    number = number + processCaseOne(arguments);
                    break;
                case <case2>:
                    number = number + processCaseTwo(arguments);
                    break;
                 ...
                case <caseN>:
                    number = number + processCaseN(arguments);
                    break;
```